.github/workflows/pull-request.yaml

```diff
@@ -1,38 +1,52 @@
 # This workflow will lint the PR's title and commit message.
-# https://github.com/CondeNast/conventional-pull-request-action
-# https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
-# https://github.com/CondeNast/conventional-pull-request-action/issues/15
 
 name: Pull Request
 on:
   pull_request:
     branches: [main, staging]
     types:
     - opened
+    - reopened
     - edited
     - synchronize
 permissions:
   contents: read
 
 jobs:
-  pr:
-    permissions:
-      pull-requests: read  # for CondeNast/conventional-pull-request-action to get specific PR & list commits
-    name: Lint PR title
+  conventional-commits:
+    name: Check PR title and commit messages
     runs-on: ubuntu-latest
     steps:
     - name: Harden Runner
       uses: step-security/harden-runner@74b568e8591fbb3115c70f3436a0c6b0909a8504
       with:
         egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
-
-    - name: Run conventional commits check
-      uses: CondeNast/conventional-pull-request-action@eef02ae6e88e4b462f39931113b1a75cc275beee
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Check out repository
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
       with:
-        commitlintRulesPath: ./commitlint.rules.js
-        commitTitleMatch: 'true'
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 # v4.1.0
+      with:
+        python-version: '3.10'
+    - name: Create and activate virtual environment
+      run: |
+        make venv
+        source .venv/bin/activate
+      env:
+        PYTHON: python
+    - name: Install dependencies
+      run: make setup
+    - name: Check PR title
+      run: echo "$PR_TITLE" | cz check
+      env:
+        PR_TITLE: ${{ github.event.pull_request.title }}
+    - name: Check PR commit messages
+      run: cz check --rev-range origin/$PR_BASE_BRANCH_NAME..origin/$PR_HEAD_BRANCH_NAME
+      env:
+        PR_BASE_BRANCH_NAME: ${{ github.event.pull_request.base.ref }}
```

Comment:
Can we inline these variables?

Reply:
I think inlining would mean to open the door to injection, see also the [**Remidiation** section in this blog](https://securitylab.github.com/research/github-actions-untrusted-input/#remediation). The initial code of this PR has these values inlined:

https://github.com/jenstroeger/python-package-template/blob/2f357e1465d2a7f252304788141b5549c41cc36a/.github/workflows/pull-request.yaml#L40-L43

With a PR title like 
```
chore: `echo test`
```
we’d have Github expand the `run:` to the (harmless but injected) shell command
```
echo chore: `echo test` | cz check 
```
That’s what we want to avoid.

Reply:
See also https://github.com/jenstroeger/python-package-template/pull/263#discussion_r923930333 above.